bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -134,6 +144,17 @@ class SourceSelector {
   uint16_t nominatedPrimaryCounter_ = 0;
   uint16_t minPrePrepareMsgsForPrimaryAwarness_ = 10;
   logging::Logger &logger_;
+
+ public:
+  // Metric
+  concordMetrics::Component metrics_component_;
```

Comment:
why do you need it as a class member?
Should be a ctor argument, no more.

Reply:
Thanks. Please let me know if its not the way you have expected.

Reply:
Now you already need it as a memeber..